src/gui/qgstablewidgetbase.h

```diff
@@ -0,0 +1,67 @@
+/***************************************************************************
+    qgstablewidgetbase.h
+     --------------------------------------
+    Date                 : 09.2016
+    Copyright            : (C) 2016 Patrick Valsecchi
+    Email                : patrick.valsecchi@camptocamp.com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSTABLEWIDGETBASE_H
+#define QGSTABLEWIDGETBASE_H
+
+#include "ui_qgstablewidgetbase.h"
+#include <QAbstractTableModel>
+#include <QVariant>
+
+/** \ingroup gui
+ * Base widget allowing to edit a collection, using a table.
```

Comment:
I'd add some extra detail here - eg "includes inbuilt buttons for adding and removing rows"